### PR TITLE
Feature/log to csv

### PR DESCRIPTION
**Pull request by yairigal**
Refactored the debug script, added CLI arguments, saved to log file and CSV file.
Added log lines for sensors samples.

### CODE REVIEW

File: algo.py [hunk 1]
```diff
@@ -287,12 +289,17 @@ def sampling_iteration(self):
...
         if seconds_from_last_breath >= 12:
             self._events.alerts_queue.enqueue_alert(AlertCodes.NO_BREATH)
+            self.log.debug('No breath')
 
         # Read from sensors
         flow_slm = self._flow_sensor.read()
         pressure_cmh2o = self._pressure_sensor.read()
         o2_saturation_percentage = self._oxygen_a2d.read()
 
+        self.log.debug('flow sensor value: %s' % flow_slm)
```
Comment (osherdp) [inline algo.py]: Maybe it should be NOTICE level as shukrun says

File: algo.py [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 from collections import deque
+
 from scipy.stats import linregress
 
 from data.alerts import AlertCodes
 from data.measurements import Measurements
 from data.configurations import Configurations
 
+NOTICE = logging.DEBUG - 1
+logging.addLevelName(NOTICE, 'NOTICE')
```
Comment (avrahamshukron) [inline algo.py]: Bad name. `NOTICE` is slightly higher than `info` in the Linux world
See https://he.wikipedia.org/wiki/Syslog#%D7%93%D7%97%D7%99%D7%A4%D7%95%D7%AA

I would suggest `TRACE`

**Comment by yairigal** [comment]
Changed the log fetching process.
Instead of sending it over socket, I access the RPi using FTP and pull the log files.
The log files then are processed in the local machine into a CSV file.

File: drivers/sfm3200_flow_sensor.py [hunk 3]
```diff
@@ -4,12 +4,17 @@
...
 
-from errors import PiGPIOInitError, I2CDeviceNotFoundError, \
-                I2CReadError, I2CWriteError, FlowSensorCRCError
+import pigpio
+
+from errors import PiGPIOInitError
+from errors import I2CDeviceNotFoundError
+from errors import I2CReadError
+from errors import I2CWriteError
+from errors import FlowSensorCRCError
```
Comment (osherdp) [inline drivers/sfm3200_flow_sensor.py:16]: Import all of them at once

File: main.py [hunk 4]
```diff
@@ -2,38 +2,18 @@
...
-        except OSError as e:
-            if e.errno != 101:
-                raise RuntimeError("Network is unreachable") from e
-
-    def makeSocket(self):
-        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-        return sock
+GB_IN_BYTES = 2 ** 30
```
Comment (osherdp) [inline main.py:16]: the opposite

File: main.py [hunk 5]
```diff
@@ -42,25 +22,21 @@ def configure_logging(level):
...
-    ch = logging.StreamHandler()
-    ch.setLevel(level)
-    # create socket handler to broadcast logs
-    sh = BroadcastHandler('255.255.255.255', config.debug_port)
-    sh.setLevel(level)
+    file_handler = RotatingFileHandler('inhalator.log',
+                                       maxBytes=GB_IN_BYTES,
+                                       backupCount=10)
+    file_handler.setLevel(level)
+    # # create console handler with a higher log level
```
Comment (osherdp) [inline main.py:29]: ?

File: scripts/save_logs.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+import datetime
+import ftplib
+import itertools
+import logging
+import re
+
+TRACE = logging.DEBUG - 1
+logging.addLevelName(TRACE, 'TRACE')
+
+RPI_IP = '192.168.43.234'
```
Comment (osherdp) [inline scripts/save_logs.py:13]: ?

Comment (yairigal) [reply scripts/save_logs.py:13]: Default Raspberry ip. Needed to access the FTP server.
For now we need to know the IP of the raspberry to access its files.

File: main.py [hunk 7]
```diff
@@ -42,25 +22,21 @@ def configure_logging(level):
...
-    fh.setLevel(logging.DEBUG)
-    # create console handler with a higher log level
-    ch = logging.StreamHandler()
-    ch.setLevel(level)
-    # create socket handler to broadcast logs
-    sh = BroadcastHandler('255.255.255.255', config.debug_port)
-    sh.setLevel(level)
+    file_handler = RotatingFileHandler('inhalator.log',
+                                       maxBytes=BYTES_IN_GB,
+                                       backupCount=10)
```
Comment (avrahamshukron) [inline main.py:27]: 10GB of logs? are you sure?

**Pull request merged**